src/plugins/data_view_editor/public/components/data_view_flyout_content_container.tsx

```diff
@@ -18,21 +18,33 @@ const IndexPatternFlyoutContentContainer = ({
   onCancel = () => {},
   defaultTypeIsRollup,
   requireTimestampField = false,
+  editData,
 }: DataViewEditorProps) => {
   const {
     services: { dataViews, notifications },
   } = useKibana<DataViewEditorContext>();
 
   const onSaveClick = async (dataViewSpec: DataViewSpec) => {
     try {
-      const indexPattern = await dataViews.createAndSave(dataViewSpec);
+      let indexPattern;
+      if (editData) {
+        const { name = '', timeFieldName, title = '' } = dataViewSpec;
+        editData.title = title;
+        editData.name = name;
+        editData.timeFieldName = timeFieldName;
+        indexPattern = await dataViews.updateSavedObject(editData);
+      } else {
+        indexPattern = await dataViews.createAndSave(dataViewSpec);
+      }
 
-      const message = i18n.translate('indexPatternEditor.saved', {
-        defaultMessage: "Saved '{indexPatternTitle}'",
-        values: { indexPatternTitle: indexPattern.title },
-      });
-      notifications.toasts.addSuccess(message);
-      await onSave(indexPattern);
+      if (indexPattern && !(indexPattern instanceof Error)) {
```

Comment:
Interesting... I wonder if the api indeed can return `Error` or if this is a mistake in types. I'd expect it to throw in case of error. @mattkime

Reply:
@shivindera As best I can tell its not possible for indexPattern to be an instanceof Error.

Reply:
@shivindera As best I can tell its not possible for indexPattern to be an instanceof Error.

Reply:
This check is there as `await dataViews.updateSavedObject(editData);` returns `Promise<void | DataView | Error>`. I have changed the variable name accordingly to reflect that.